src/progress/Progress.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  useProgressState,
+  ProgressInitialState,
+  ProgressStateReturn,
+} from "@renderlesskit/react";
+import * as React from "react";
+
+import { BoxProps } from "../box";
+import { ProgressBar } from "./ProgressBar";
+import { ProgressTrack } from "./ProgressTrack";
+import { createContext, runIfFn } from "../utils";
+import { forwardRefWithAs } from "../utils/types";
+
+export type ProgressContext = {
+  state: ProgressStateReturn;
+  size: ProgressProps["size"];
+};
+
+const [ProgressProvider, useProgress] = createContext<ProgressContext>({
```

Comment:
Rename `useProgress` to `useProgressContext` because people might get confuse thinking that useProgress is a state hook.